### YamlMime:JavaType
uid: "com.azure.core.exception.ClientAuthenticationException"
fullName: "com.azure.core.exception.ClientAuthenticationException"
name: "ClientAuthenticationException"
nameWithType: "ClientAuthenticationException"
summary: "The exception thrown when failed to authenticate the client request with status code of 4XX, typically 401 unauthorized. A runtime exception indicating request authorization failure caused by one of the following scenarios:\n\n *  A client did not send the required authorization credentials to access the requested resource, i.e. Authorization HTTP header is missing in the request\n *  If the request contains the HTTP Authorization header, then the exception indicates that authorization has been refused for the credentials contained in the request header."
inheritances:
- "<xref href=\"java.lang.Object\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Throwable\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Exception\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.RuntimeException\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.exception.AzureException\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.exception.HttpResponseException\" data-throw-if-not-resolved=\"False\" />"
inheritedMembers:
- "com.azure.core.exception.HttpResponseException.getResponse()"
- "com.azure.core.exception.HttpResponseException.getValue()"
- "java.lang.Object.clone()"
- "java.lang.Object.equals(java.lang.Object)"
- "java.lang.Object.finalize()"
- "java.lang.Object.getClass()"
- "java.lang.Object.hashCode()"
- "java.lang.Object.notify()"
- "java.lang.Object.notifyAll()"
- "java.lang.Object.wait()"
- "java.lang.Object.wait(long)"
- "java.lang.Object.wait(long,int)"
- "java.lang.Throwable.addSuppressed(java.lang.Throwable)"
- "java.lang.Throwable.fillInStackTrace()"
- "java.lang.Throwable.getCause()"
- "java.lang.Throwable.getLocalizedMessage()"
- "java.lang.Throwable.getMessage()"
- "java.lang.Throwable.getStackTrace()"
- "java.lang.Throwable.getSuppressed()"
- "java.lang.Throwable.initCause(java.lang.Throwable)"
- "java.lang.Throwable.printStackTrace()"
- "java.lang.Throwable.printStackTrace(java.io.PrintStream)"
- "java.lang.Throwable.printStackTrace(java.io.PrintWriter)"
- "java.lang.Throwable.setStackTrace(java.lang.StackTraceElement[])"
- "java.lang.Throwable.toString()"
syntax: "public class ClientAuthenticationException extends HttpResponseException"
constructors:
- "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(java.lang.String,com.azure.core.http.HttpResponse)"
- "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(java.lang.String,com.azure.core.http.HttpResponse,java.lang.Object)"
- "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(java.lang.String,com.azure.core.http.HttpResponse,java.lang.Throwable)"
type: "class"
metadata: {}
package: "com.azure.core.exception"
artifact: com.azure:azure-core:1.3.0-beta.1
